### PR TITLE
Refactor components redux

### PR DESCRIPTION
**Pull request by boyntoni**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,5 +18,10 @@
     "webpack": "2.2.1",
     "webpack-dev-server": "^2.9.7",
     "write-file-webpack-plugin": "3.4.2"
+  },
+  "dependencies": {
+    "vue": "^2.5.13",
+    "vue-loader": "^13.6.0",
+    "vue-template-compiler": "^2.5.13"
```
Comment (dYale) [inline package.json]: vue-loader
vue-template-compiler 

These should be dev-dependencies

File: src/js/components/troubleshooting/ads-debug.vue [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        copyAdTag(text) {
+            var copyText = document.createElement('input');
+            copyText.type = 'text';
+            document.body.appendChild(copyText);
+            copyText.style = 'display: inline; width: 1px;';
+            copyText.value = text;
+            copyText.focus();
+            document.execCommand('SelectAll');
+            document.execCommand('Copy');
+        },
```
[the remark this answers is not in the document]
Comment (boyntoni) [reply src/js/components/troubleshooting/ads-debug.vue]: I ended up changing this entirely. I ended up using an NPM module for this; there is something funky about using the doc.execCommand with Vue. The NPM module is 2kb though.

File: src/js/components/general/event-detail.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    },
+    methods: {
+        formattedValue(value) {
+            if (typeof value && typeof value === 'number' && value.length > 5) {
+                return value.toFixed(5);
+            }
+            return value;
+        },
+        isStringOrShortInt(value) {
```
[the remark this answers is not in the document]
Comment (boyntoni) [reply src/js/components/general/event-detail.vue]: Changed and will reflect in future PR

File: src/js/components/general/event-detail.vue [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    props: ['activeEvent', 'parentComponent'],
+    computed: {
+        apiUrl() {
+            let type = this.activeEvent.type;
+            return 'https://developer.jwplayer.com/jw-player/docs/javascript-api-reference/#jwplayeron' +  type.toLowerCase(); 
+        }
+    },
+    methods: {
+        formattedValue(value) {
+            if (typeof value && typeof value === 'number' && value.length > 5) {
```
[the remark this answers is not in the document]
Comment (boyntoni) [reply src/js/components/general/event-detail.vue]: Changed to be `value` in future PR

File: src/js/components/troubleshooting/ads-debug.vue [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        filterAdErrors() {
+            let adErrors = this.allEvents.filter(event => event.type == 'adError');
+            if (!!adErrors.length) {
+                this.adErrors = adErrors;
```
[the remark this answers is not in the document]
Comment (boyntoni) [reply src/js/components/troubleshooting/ads-debug.vue]: But if: 

x = []
!!x.length => false 

x=[{}]
!!x.length => true

Also, this has been slightly modified in a future PR as well to be a computed value, which achieves a better result as far as the component updating in real-time as more adErrors are included.

**Pull request merged**